bpf/lib/policy.h

```diff
@@ -24,14 +24,37 @@
 #ifdef POLICY_ENFORCEMENT
 
 static inline int policy_can_access(void *map, struct __sk_buff *skb, __u32 src_label,
-				    size_t cidr_addr_size, void *cidr_addr)
+				    __u16 dport, __u8 proto, size_t cidr_addr_size, void *cidr_addr)
 {
 #ifdef DROP_ALL
 	return DROP_POLICY;
 #else
 	struct policy_entry *policy;
 
-	policy = map_lookup_elem(map, &src_label);
+	struct policy_key key = {
+		.sec_label = src_label,
+		.dport = dport,
+		.protocol = proto,
+		.pad = 0,
+	};
+
+#ifdef HAVE_L4_POLICY
+	policy = map_lookup_elem(map, &key);
+	if (likely(policy)) {
+		cilium_dbg3(skb, DBG_L4_CREATE, src_label, SECLABEL,
+			    dport << 16 | proto);
+
+		/* FIXME: Use per cpu counters */
+		__sync_fetch_and_add(&policy->packets, 1);
+		__sync_fetch_and_add(&policy->bytes, skb->len);
+		return TC_ACT_OK;
+	}
```

Comment:
Should we consider tracing either the positive / negative match cases for L4 policy?

Reply:
I think what you have is fine to get started.